src/cmd/Makefile.am

```diff
@@ -21,7 +21,6 @@ fluxcmd_ldadd = \
 	$(top_builddir)/src/common/libflux-optparse.la \
 	$(FLUX_SECURITY_LIBS) \
 	$(LIBPTHREAD) \
-	$(LIBDL) \
```

Comment:
flux-module.c includes dlcfn.h but doesn't appear to use it.  Maybe drop that too?